mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala

```diff
@@ -519,6 +519,12 @@ class SparseMatrix(
       rowIndices: Array[Int],
       values: Array[Double]) = this(numRows, numCols, colPtrs, rowIndices, values, false)
 
+  override def equals(o: Any): Boolean = o match {
+    case m: Matrix =>
+      m.numRows == numRows && m.numCols == numCols && Arrays.equals(toArray, m.toArray)
```

Comment:
I like what you had before better.  Using Breeze, I think you'd avoid converting to dense representations.  With toArray, it becomes dense.